### PR TITLE
Implement permit functionality 

### PR DESCRIPTION
**Pull request by haythem96**
# Implement permit functionality 

## High Level Description

This PR add the `permit()` function to `Otoken.sol` using the `ERC20PermitUpgradeable.sol` from OZ package. 

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/packages/oz/upgradeability/utils/CountersUpgradeable.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * @author Matt Condon (@shrugs)
+ * @dev Provides counters that can only be incremented or decremented by one. This can be used e.g. to track the number
+ * of elements in a mapping, issuing ERC721 ids, or counting request ids.
+ *
+ * Include with `using Counters for Counters.Counter;`
+ * Since it is not possible to overflow a 256 bit integer with increments of one, `increment` can skip the {SafeMath}
+ * overflow check, thereby saving gas. This does assume however correct usage, in that the underlying `_value` is never
+ * directly accessed.
+ */
+library CountersUpgradeable {
```
Comment (antoncoding) [inline contracts/packages/oz/upgradeability/utils/CountersUpgradeable.sol:18]: Do we need this? 
Don't see this library being used anywhere. (Ctrl + F found nothing 😶  )

**Review by antoncoding — APPROVED** [review]
Lol the change on oToken.sol is really just 1 line

File: contracts/packages/oz/upgradeability/utils/CountersUpgradeable.sol (hunk 1 above)
Comment (haythem96) [reply contracts/packages/oz/upgradeability/utils/CountersUpgradeable.sol:18]: It is in `ERC20PermitUpgradeable.sol`

**Review by aparnakr — APPROVED** [review]
Looks good, is there a way to test the permit functionality?

**Comment by haythem96** [comment]
> Looks good, is there a way to test the permit functionality?

Didn't add tests here as I used OZ contracts. 
Will add a tests for the permit() call in the PermitCallee

**Pull request merged**